app/uk/gov/hmrc/exports/models/declaration/notifications/ParsedNotification.scala

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.exports.models.declaration.notifications
+
+import play.api.libs.functional.syntax._
+import play.api.libs.json.{Json, Reads, _}
+import reactivemongo.bson.BSONObjectID
+
+case class ParsedNotification(id: BSONObjectID = BSONObjectID.generate(), actionId: String, payload: String, details: NotificationDetails)
```

Comment:
Is there a good reason to keep NotificationDetails as a separate class instead of just merging the fields directly into the ParsedNotification class? Is 'payload' needed here (repetition of verbose textual data which at this stage is redundant)?

Reply:
The payload field has to be there for now, because frontend service expects it (it is not optional). But I want to change the frontend model and remove this field here in the next PR.

As for the NotificationDetails, I consider all other fields as kind of metadata and `details` field is the relevant data. Besides, having flat structure is more susceptible to information noise. Right now the data is more structured and so extending the class with new fields does not mix purposes of these (whether it's data, or some helper piece of information field).